backend/selectgen.ml

```diff
@@ -174,6 +174,7 @@ let oper_result_type = function
   | Calloc _ -> typ_val
   | Cstore (_c, _) -> typ_void
   | Cprefetch _ -> typ_void
+  | Catomic _ -> typ_int
```

Comment:
`Ifetch_and_add` is `typ_int` but `Icompare_and_swap` should be `typ_val` now that it returns a tagged bool.